File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/Sample Code/Join.py
## Function to join automotive data
def auto_join(autos, makes):
    import pandas as pd
    key = ['make-id'] ## Define key column
    ## Return the joined dataframe
    return pd.merge(autos, makes, on = key, how = 'left')

## Perform the join in AML    
def azureml_main(autos, makes):
    return auto_join(autos, makes)

File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/Sample Code/Join.R
## Define a function to join the data frames
join.auto <- function(autos, makes){
  require(dplyr) ## Make sure dplyr is loaded
  left_join(autos, makes, by = 'make-id')
}

## R code to join the two input tables
autos <- maml.mapInputPort(1) # read autos data frame from port 1
makes <- maml.mapInputPort(2) # read makes data frame from port 2
out <- join.auto(autos, makes) ## Join the data frames
maml.mapOutputPort("out") ## Output the joined data frame
